pkg/registry/event/registry.go

```diff
@@ -47,6 +47,10 @@ func NewEtcdRegistry(h tools.EtcdHelper, ttl uint64) generic.Registry {
 			TTLFunc: func(runtime.Object, bool) (uint64, error) {
 				return ttl, nil
 			},
+			DeleteCollectionFunc: func(ctx api.Context) (runtime.Object, error) {
+				// TODO Create and return status object
+				return nil, h.Delete(etcdgeneric.NamespaceKeyRootFunc(ctx, prefix), true)
```

Comment:
Can you define a common function that all namespaced resources can use as the ```DeleteCollectionFunc``.

For example:

```code
func DeleteCollectionPerNamespaceFunc func(...) ... {
  return &api.Status{...}, ...
}
```